kong/init.lua

```diff
@@ -545,10 +521,12 @@ function Kong.init()
   end
 
   if config.database == "off" then
-    local err
-    declarative_entities, err, declarative_meta = parse_declarative_config(kong.configuration)
-    if not declarative_entities then
-      error(err)
+    if is_http_module or #config.proxy_listeners == 0 then
```

Comment:
Need to check `admin_listeners` or `status_listeners` as well.

Reply:
not sure for status_listners as it's readonly

Reply:
We will follow what the `kong/templates/nginx.lua` does as that ultimately determines whether `http`'s `init` phase is rendered or not.